app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/topsites/TopSitesPagerAdapter.kt

```diff
@@ -10,31 +10,60 @@ import androidx.recyclerview.widget.DiffUtil
 import androidx.recyclerview.widget.ListAdapter
 import kotlinx.android.synthetic.main.component_top_sites.view.*
 import mozilla.components.feature.top.sites.TopSite
+import org.mozilla.fenix.home.sessioncontrol.AdapterItem
 import org.mozilla.fenix.home.sessioncontrol.TopSiteInteractor
 import org.mozilla.fenix.home.sessioncontrol.viewholders.TopSiteViewHolder
 
 class TopSitesPagerAdapter(
     private val interactor: TopSiteInteractor
-) : ListAdapter<List<TopSite>, TopSiteViewHolder>(DiffCallback) {
+) : ListAdapter<List<TopSite>, TopSiteViewHolder>(TopSiteListDiffCallback) {
 
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TopSiteViewHolder {
         val view = LayoutInflater.from(parent.context)
             .inflate(TopSiteViewHolder.LAYOUT_ID, parent, false)
         return TopSiteViewHolder(view, interactor)
     }
 
+    override fun onBindViewHolder(
+        holder: TopSiteViewHolder,
+        position: Int,
+        payloads: MutableList<Any>
+    ) {
+        if (payloads.isNullOrEmpty()) {
+            onBindViewHolder(holder, position)
+        } else {
+            if (payloads[0] is AdapterItem.TopSitePagerPayload) {
+                val adapter = holder.itemView.top_sites_list.adapter as TopSitesAdapter
+                val payload = payloads[0] as AdapterItem.TopSitePagerPayload
+                for (item in payload.changed) {
+                    adapter.notifyItemChanged(item.first, item.second)
+                }
+            }
+        }
+    }
+
     override fun onBindViewHolder(holder: TopSiteViewHolder, position: Int) {
         val adapter = holder.itemView.top_sites_list.adapter as TopSitesAdapter
         adapter.submitList(getItem(position))
     }
 
-    private object DiffCallback : DiffUtil.ItemCallback<List<TopSite>>() {
+    object TopSiteListDiffCallback : DiffUtil.ItemCallback<List<TopSite>>() {
```

Comment:
I think internal or package-private could have also worked if this was just for testing.